tmt/checks/watchdog.py

```diff
@@ -0,0 +1,467 @@
+import dataclasses
+import datetime
+import re
+import threading
+import time
+from collections.abc import Iterable
+from typing import TYPE_CHECKING, Optional
+
+import tmt.log
+import tmt.steps.execute
+import tmt.steps.provision
+import tmt.steps.provision.artemis
+import tmt.steps.provision.connect
+import tmt.steps.provision.local
+import tmt.steps.provision.mrack
+import tmt.steps.provision.podman
+import tmt.steps.provision.testcloud
+import tmt.utils
+from tmt.checks import Check, CheckPlugin, provides_check
+from tmt.result import CheckResult, ResultOutcome
+from tmt.utils import Path, field, render_run_exception_streams
+
+if TYPE_CHECKING:
+    from tmt.steps.execute import TestInvocation
+
+PING_OUTPUT_PATTERN = re.compile(
+    r'(?m)(?P<transmitted>\d+) packets transmitted, (?P<received>\d+) received')
+SSH_PING_OUTPUT_PATTERN = re.compile(r'Ncat: Connected')
+
+PINGABLE_GUEST_CLASSES: tuple[type[tmt.steps.provision.Guest], ...] = (
+    tmt.steps.provision.artemis.GuestArtemis,
+    tmt.steps.provision.connect.GuestConnect,
+    tmt.steps.provision.mrack.GuestBeaker,
+    # TODO: is there a way to ping the VM instead of localhost?
```

Comment:
`qemu:///system` (`tmt run ... provision -h virtual -c system ...`) assigns IP but one has to be in libvirt group to use it

Reply:
I tried exactly that in the test, https://github.com/teemtee/tmt/pull/2412/files#diff-147638a559c0bc22967b8321bbb808b7124063991f148d13e0d08997f7c0efd8R34, but still I was unable to simulate "dead guest" on ICMP level. Despite disabling all ICMP traffic in the guest, I was still getting responses...

But I guess this is not critical. Unfortunate, yes, but I was able to test with `virtual` and `ssh-ping`, and I plan to extend the test to Beaker and Artemis in the future.